File: java/pr9/src/main/java/com/company/pr9/entity/Bus.java
package com.company.pr9.entity;

import io.jmix.core.entity.annotation.JmixGeneratedValue;
import io.jmix.core.metamodel.annotation.InstanceName;
import io.jmix.core.metamodel.annotation.JmixEntity;

import javax.persistence.*;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import java.util.UUID;
import java.util.List;
import java.util.Date;

@JmixEntity
@Table(name = "TR_BUS")
@Entity(name = "tr_Bus")
public class Bus {
    @JmixGeneratedValue
    @Column(name = "ID", nullable = false)
    @Id
    private UUID id;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    @Column(name = "MODEL")
    private String model;

    public String getModel() {
        return model;
    }

    public void setModel(String model) {
        this.model = model;
    }

    @InstanceName
    @Column(name = "REGISTRATION_NUMBER")
    private String registrationNumber;

    public String getRegistrationNumber() {
        return registrationNumber;
    }

    public void setRegistrationNumber(String registrationNumber) {
        this.registrationNumber = registrationNumber;
    }

    @Column(name = "SEAT_COUNT")
    private Integer seatCount;

    public Integer getSeatCount() {
        return seatCount;
    }

    public void setSeatCount(Integer seatCount) {
        this.seatCount = seatCount;
    }

    @Column(name = "START_DATE")
    private Date startDate;

    public Date getStartDate() {
        return startDate;
    }

    public void setStartDate(Date startDate) {
        this.startDate = startDate;
    }

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "STATION_ID")
    private Station station;

    public Station getStation() {
        return station;
    }

    public void setStation(Station station) {
        this.station = station;
    }

    @ManyToMany
    @JoinTable(name = "BUS_ROUTE_LINK",
            joinColumns = @JoinColumn(name = "BUS_ID", referencedColumnName = "ID"),
            inverseJoinColumns = @JoinColumn(name = "ROUTE_ID", referencedColumnName = "ID"))
    private List<Route> routes;

    public List<Route> getRoutes() {
        return routes;
    }

    public void setRoutes(List<Route> routes) {
        this.routes = routes;
    }

}

File: java/pr9/src/main/kotlin/com/company/pr9/entity/Bus.kt
package com.company.pr9.entity

import io.jmix.core.entity.annotation.JmixGeneratedValue
import io.jmix.core.metamodel.annotation.InstanceName
import io.jmix.core.metamodel.annotation.JmixEntity
import java.util.*
import javax.persistence.*

@JmixEntity
@Table(name = "BUS")
@Entity
open class Bus {
    @JmixGeneratedValue
    @Column(name = "ID", nullable = false)
    @Id
    private var internalId: String? = null

    fun getId(): String? {
        return internalId
    }

    fun setId(id: String?) {
        this.internalId = id
    }
    @InstanceName
    @Column(name = "MODEL")
    private val model: String? = null

    @Column(name = "REGISTRATION_NUMBER")
    private val registrationNumber: String? = null

    @Column(name = "SEATING_CAPACITY")
    private val seatingCapacity = 0

    @Temporal(TemporalType.DATE)
    @Column(name = "START_DATE")
    private val startDate: Date? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "STATION_ID")
    private val station: Station? = null

    @ManyToMany
    @JoinTable(
        name = "BUS_ROUTE_LINK",
        joinColumns = [JoinColumn(name = "BUS_ID")],
        inverseJoinColumns = [JoinColumn(name = "ROUTE_ID")]
    )
    private val routes: List<Route> = ArrayList<Route>()

    // Constructors, getters, setters
}
